pkg/storage/testdata/mvcc_histories/gc_clear_full_range

```diff
@@ -0,0 +1,98 @@
+run ok
+put k=A v=B ts=10
+----
+>> at end:
+data: "A"/10.000000000,0 -> /BYTES/B
+
+# Check that we can't invoke gc_clear_range if we have live data.
```

Comment:
Let's add a couple of variants that check a live value and a point tombstone that are exactly on the GC threshold too (or do we have them already?).

Reply:
Not much sense for the full range deletion. I add one more test but overall we bail if there's anything not masked by range tombstone even if technically it should be collectable. E.g. point tombstone followed by range tombstone is not enough for fast path to succeed.

Reply:
The motivation is that I want to make sure we have test coverage for this branch:

https://github.com/cockroachdb/cockroach/pull/90830/files#diff-2a418618ad51e1cdef4b44240ef9ee4789e9f97f2d6d9734b67ea7fc94d5d9afR5452-R5457

I don't believe we currently have any test cases that hit it?

Reply:
Actually I think this condition is not needed at all. If key is garbage then it must be covered by newer value (its timestamp must be below coveredBy). And coveredBy timestamp is guaranteed to be at or below gcThreshold so we must be good.